[wasm] Simd refactoring

radekdoulik opened the pull request:
Rename few opcodes to contain `_SIMD_`.

Use more SimdIntrinsic fields to simplify the code.

radekdoulik committed:
[wasm] Simd refactoring

Rename few opcodes to contain `_SIMD_`.

Use more SimdIntrinsic fields to simplify the code.

radekdoulik committed:
Fix build

radekdoulik committed:
Fix narrowing intrinsics

vargaz commented on src/mono/mono/mini/simd-intrinsics.c:
```diff
@@ -4711,8 +4711,8 @@ static SimdIntrinsic packedsimd_methods [] = {
 	{SN_CompareNotEqual},
-	{SN_ConvertNarrowingSignedSaturate, OP_XOP_X_X_X},
-	{SN_ConvertNarrowingUnsignedSaturate, OP_XOP_X_X_X},
+	{SN_ConvertNarrowingSignedSaturate, OP_XOP_X_X_X, 0},
```
This is probably not needed, the unset fields are 0.

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik committed:
Verbose aot compilation

radekdoulik committed:
Feedback

radekdoulik committed:
Make it verbose on helix

kg commented on src/mono/mono/mini/mini-ops.h:
```diff
@@ -1757,17 +1757,17 @@ MINI_OP3(OP_ARM64_SQRDMLSH_SCALAR, "arm64_sqrdmlsh_scalar", XREG, XREG, XREG, XR
+MINI_OP(OP_SIMD_SSHR, "sshr", XREG, XREG, IREG)
+MINI_OP(OP_SIMD_USHR, "ushr", XREG, XREG, IREG)
+MINI_OP3(OP_SIMD_USRA, "usra", XREG, XREG, XREG, IREG)
+MINI_OP3(OP_SIMD_SSRA, "ssra", XREG, XREG, XREG, IREG)
```
shouldn't the string versions also start with simd_? like for the wasm ones below

kg commented on src/mono/mono/mini/simd-intrinsics.c:
```diff
@@ -4852,25 +4849,16 @@ emit_wasm_supported_intrinsics (
-			case SN_Swizzle:
-				return emit_simd_ins_for_sig (cfg, klass, OP_WASM_SIMD_SWIZZLE, -1, -1, fsig, args);
 		}
+		if (op != 0)
```
Maybe add a comment specifying that this is a default fallback

kg commented on src/mono/mono/mini/simd-intrinsics.c:
```diff
@@ -4796,54 +4796,51 @@ emit_wasm_supported_intrinsics (
-				if (intrins != -1)
-						return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_X, intrins, arg0_type, fsig, args);
+				if (c0 != 0)
+						break;
```
Might also be good to have comments indicating that these breaks are explicitly intended to use the fallback, like a fall-through

radekdoulik committed:
Do not set op for narrowing methods

Because `emit_hardware_intrinsics` doesn't call custom emit in such case
and so we endup with code emitted with wrong zero c0.

radekdoulik committed:
Feedback

radekdoulik committed:
Be quiet again :-)

radekdoulik commented:
The failing builds are unrelated
